src/components/accession2/view/ProjectSelectorModal.tsx

```diff
@@ -0,0 +1,116 @@
+import { makeStyles } from '@mui/styles';
+import strings from 'src/strings';
+import Button from 'src/components/common/button/Button';
+import DialogBox from 'src/components/common/DialogBox/DialogBox';
+import { useProjects } from 'src/components/InventoryV2/form/useProjects';
+import { Accession } from 'src/types/Accession';
+import ProjectsDropdown from 'src/components/InventoryV2/form/ProjectsDropdown';
+import { useAppDispatch, useAppSelector } from 'src/redux/store';
+import { selectProjectRequest } from 'src/redux/features/projects/projectsSelectors';
+import useSnackbar from 'src/utils/useSnackbar';
+import { requestProjectAssign } from 'src/redux/features/projects/projectsAsyncThunks';
+
+const useStyles = makeStyles((theme: Theme) => ({
+  addIcon: {
+    fill: theme.palette.TwClrIcnBrand,
+    height: '20px',
+    width: '20px',
+    marginRight: '5px',
+  },
+}));
+
+interface ProjectSelectorModalProps {
+  accession: Accession;
+}
+
+const ProjectSelectorModal = (props: ProjectSelectorModalProps) => {
+  const dispatch = useAppDispatch();
+  const theme = useTheme();
+  const classes = useStyles();
+  const snackbar = useSnackbar();
+
+  const { availableProjects } = useProjects(props.accession);
+
+  const [isOpen, setIsOpen] = useState<boolean>(false);
+  const [accession, setAccession] = useState(props.accession);
+
+  const [requestId, setRequestId] = useState('');
+  const projectRequest = useAppSelector((state) => selectProjectRequest(state, requestId));
+
+  const handleCancel = useCallback(() => {
+    setIsOpen(false);
+  }, []);
+
+  const handleSave = useCallback(() => {
+    if (accession.projectId && props.accession.projectId !== accession.projectId) {
+      const request = dispatch(
+        requestProjectAssign({ projectId: accession.projectId, entities: { accessionIds: [accession.id] } })
+      );
+      setRequestId(request.requestId);
+    }
+
+    setIsOpen(false);
+  }, [accession, dispatch, props.accession.projectId]);
+
+  const handleUpdateProject = useCallback(
+    (setFn: (previousAccession: Accession) => Accession) => {
+      const nextAccession = setFn(accession);
+      setAccession(nextAccession);
+    },
+    [accession]
+  );
+
+  useEffect(() => {
+    if (projectRequest?.status === 'success') {
+      setRequestId('');
+    } else if (projectRequest?.status === 'error') {
+      snackbar.toastError(strings.GENERIC_ERROR);
+    }
+
+    return () => setRequestId('');
+  }, [projectRequest, snackbar]);
+
+  const linkStyle = {
+    color: theme.palette.TwClrTxtBrand,
+    fontWeight: 500,
+    textDecoration: 'none',
+    cursor: 'pointer',
+    fontSize: '14px',
+  };
+
+  return (
+    <>
+      <LinkMUI sx={linkStyle} onClick={() => setIsOpen(true)}>
+        <Box display='flex' alignItems='center'>
+          <Icon name='iconAdd' className={classes.addIcon} />
+          {strings.ADD_TO_PROJECT}
+        </Box>
+      </LinkMUI>
```

Comment:
Would it make sense to put the modal trigger (and open/close) state in the parent component? To decouple the access pattern from the modal itself (maybe doesn't make sense, if we're not expecting to use this modal elsewhere or access it differently)? It's a tradeoff, making it self-contained here vs. spreading its "state" across this and the parent component. So I'd be okay either way.

Reply:
I could go either way. I did this intentionally so the component can be re-used for different parents. Since there are 3 entities that can be assigned to a project, presumably we will implement this functionality elsewhere. If we bake it into the parent, we will have more duplicate code when we inevitably implement this modal elsewhere or we will need to refactor more stuff. 

I will make this a bit more abstract to illuminate what I'm proposing, then we can decide.

Reply:
@charlie-tf take a look now and let me know what you think, along with https://github.com/terraware/terraware-web/pull/1912. Hopefully it makes a bit more sense what I was going for now that it is more generic.

Reply:
Okay, yeah, this is good. Thanks